controllers/managedClusterResources.go

```diff
@@ -637,3 +638,20 @@ func (r *ClusterGroupUpgradeReconciler) deleteDependenciesViews(
 	}
 	return nil
 }
+
+// do the same for precaching
+func SetReasonMessage(clusterStatus map[string]string, state string) (string, string) {
+
+	var reason, message string
+	for _, state := range clusterStatus {
+		if state != state {
```

Comment:
Isn't this going to cause a problem by reusing the input variable name here? Won't this just always return false as a result?

Reply:
You are right, completely missed that in a hurry, fixed